### ISSUE TITLE
set class_weight for LightGBM in sklearn

### ISSUE THREAD

**Issue by dzliu** [Microsoft/LightGBM#1107]
Is there a way to set class weight for multi-class classification task when using LightGBM with sklearn? From reading this past issue https://github.com/Microsoft/LightGBM/issues/434, it seems like a weight file can be used to set weight per each data. I'm using LightGBM as one stage of a complex sklearn pipeline, the input data has been transformed multiple times before fed into LightGBM, thus, it is not straightforward to make LightGBM consume a weight file. Is there any way to specify class_weight as a parameter to LightGBM in a similar fashion as sklearn's logistic regression class_weight parameter?

**Comment by wxchan** [comment]
I think it won't be hard to set sample weights.

Something like `sample_weight=[class_weight[l] for l in label]`.

**Comment by StrikerRUS** [comment]
@wxchan Since this feature seems to be needed among users, I think we can implement corresponding parameter. I can took it if you don't mind.

**Comment by wxchan** [comment]
@StrikerRUS what I concern is the priority of class_weight and sample_weight, it might make user confused.

**Comment by StrikerRUS** [comment]
@wxchan Yeah, I thought about it. My variant: by default both `class_weight` and `sample_weight` are `None`. In case of both of them are not `None` raise exception and tell user use one of them. In case of binary classification and set `class_weight` also raise exception and tell user use `scale_pos_weight` or `scale_pos_weight`.

**Comment by guolinke** [comment]
@StrikerRUS @wxchan 
I think the final weight can be the multiplication of class_weight and sample_weight.

**Comment by StrikerRUS** [comment]
@guolinke Should we allow using `class_weight` in binary classification task?

**Comment by StrikerRUS** [comment]
I've created PR #1114, so let's move the discussion there.
